clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -42,20 +49,20 @@ int16_t3 abs(int16_t3);
 _HLSL_AVAILABILITY(shadermodel, 6.2)
 _HLSL_BUILTIN_ALIAS(__builtin_elementwise_abs)
 int16_t4 abs(int16_t4);
+#endif
 
-_HLSL_AVAILABILITY(shadermodel, 6.2)
+_HLSL_16BIT_AVAILABILITY(shadermodel, 6.2)
```

Comment:
Currently in dxc the16-bit types are only available when the `-enable-16bit-types` command line flag is provided. They are not available by default in SM 6.2+.

Is this availability macro making the 16-bit types always available when targeting SM 6.2+?

Reply:
Ah ok. So this isn't changing how the enabling of 16bit types are working. It just lets the 32-bit half overloads work in earlier shader models.